tests/widget/modal/widget-modal.show-password-checkbox.test.js

```diff
@@ -0,0 +1,36 @@
+import { expect, test } from '@playwright/test';
+
+import { asyncEvents } from '../../utilities/async-events.js';
+
+test('Modal widget with show password checkbox', async ({ page }) => {
+  const { clickButton, navigate } = asyncEvents(page);
+
+  const messageArray = [];
+
+  // Listen for events on page
+  page.on('console', async (msg) => {
+    return messageArray.push(msg.text());
+  });
+
+  // Navigate to page without Widget instantiation
+  await navigate('widget');
+
+  const loginLink = page.getByRole('link', { name: 'Login via Modal Widget' });
+
+  await expect(loginLink).toBeVisible();
+
+  await navigate('widget/modal?journey=TEST_Login');
+
+  await expect(page.getByRole('dialog')).toBeHidden();
+
+  await clickButton('Open Login Modal', '/authenticate');
+
+  await expect(page.getByRole('dialog')).toBeVisible();
+
+  await page.getByLabel('Username').fill('user');
+  await page.getByLabel('Password').fill('password1');
+
+  await page.getByRole('checkbox').check();
+  await expect(page.getByRole('checkbox')).toBeFocused();
+  await expect(page.getByLabel('Password')).toHaveValue('password1');
```

Comment:
This isn't technically testing the checkbox functionality. The value itself isn't going to change relative to the show or hide password feature. What changes is the `type` attribute of the `input`, so we should probably test that, instead of the value the input is holding.